### ISSUE TITLE
Build failure on Haiku

### ISSUE THREAD

**Issue by asomers** [Stebalien/tempfile#246]
Tempfile 3.7.0 fails to build on Haiku:
```
> cargo check -Z build-std --target x86_64-unknown-haiku
    Checking tempfile v3.7.0 (/usr/home/somers/src/rust/tempfile)
error[E0432]: unresolved import `rustix::fs::CWD`
  --> src/file/imp/unix.rs:17:55
   |
17 | use rustix::fs::{linkat, renameat, unlinkat, AtFlags, CWD};
   |                                                       ^^^ no `CWD` in `fs`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `tempfile` (lib) due to previous error
```
v3.6.0 worked just fine.  The problem seems to be that Rustix 0.38.0 removed `CWD`.
See https://github.com/bytecodealliance/rustix/commit/23af8a8c4ca37f6fdf602a26a67d3e32a1f64e80#diff-9f7a9943fff7afdd02920889c8f84454a8968e2f8015fd1ae06742255f82fe85 .
I suggest removing the `persist` method on Haiku.  That should be sufficient to get compilation working again.

**Comment by Stebalien** [comment]
That commit looks like it's _adding_ support for CWD.

**Comment by Stebalien** [comment]
Ah. They moved the line that disabled it.

**Comment by sunfishcode** [comment]
Yeah; Haiku's `AT_FDCWD` is -1, which can't currently be stored in a `BorrowedFd`.

I submitted https://github.com/Stebalien/tempfile/pull/247 as a possible fix here.

**Comment by Stebalien** [comment]
fixed by #247

**Comment by Stebalien** [comment]
(3.7.1 has been released)
